server/routes/profileRouter.js

```diff
@@ -1,13 +1,20 @@
 const express = require("express");
 const router = express.Router();
 const profileController = require("../controllers/profileController");
+const { validateProfile, validateSitterProfile } = require("../validate");
 // uploadController doesn't exist in the current main branch.
 // const uploadController = require("../controllers/uploadController");
-
-router.post("/", profileController.createProfile);
-router.put("/update", profileController.updateProfile);
-router.get("/", profileController.findProfile);
-router.get("/all", profileController.getAllProfiles);
+router.post("/", validateProfile, profileController.createProfile);
+router.put(
+  "/:profileId",
+  validateProfile,
+  validateSitterProfile,
+  profileController.updateProfile
+);
+router.patch("/:profileId", profileController.patchProfile);
```

Comment:
Do you have two routes for just updating profile?

Reply:
hi @ezeilo-su, one was there(PUT) already but that one expects all props and replaces the model, I created one that only updates the provided props(ex: availability). Should there only be one?

Reply:
@iamalx It is okay to have two separate routes. However, it is a good practice to consider extensibility/reusability when designing routes (having only one route would be more ideal) 😄

Reply:
okay, I created one because it suited my needs better for this issue. I didnt think it was necessary to provide the whole model to update one prop.